widget/struct_form.go

```diff
@@ -0,0 +1,146 @@
+package widget
+
+import (
+	"reflect"
+	"regexp"
+	"strings"
+	"time"
+
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/container"
+	"fyne.io/fyne/v2/data/binding"
+	"fyne.io/fyne/v2/widget"
+)
+
+type StructForm struct {
+	widget.BaseWidget
+	structType reflect.Type
+	modal      *widget.PopUp
+	canvas     fyne.Canvas
+	fields     []reflect.StructField
+	widgets    []*widget.FormItem
+	bindings   []binding.DataItem
+	submit     func(s interface{})
+}
+
+func NewStructForm(c fyne.Canvas, s interface{}, submit func(s interface{})) *StructForm {
```

Comment:
It could be useful to split the function in two, one to create just the form and another to pack it in a dialog.